monitor/ui/src/test/java/org/datacleaner/monitor/RESTEndpointsIT.java

```diff
@@ -0,0 +1,101 @@
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.monitor;
+
+import static io.restassured.RestAssured.basic;
+import static io.restassured.RestAssured.given;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.http.HttpStatus;
+import org.datacleaner.test.MonitorRestEndpoint;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+
+import io.restassured.RestAssured;
+
+public class RESTEndpointsIT {
+
+    private static final String JOBS_PATH = "/jobs/";
+    private static final String REFERENCEDATA_PATH = "/referencedata/dictionary/";
+
+    private static final int ONE_MINUTE = 60000;
+
+    private static final String USER_NAME = "admin";
+    private static final String USER_PASSWORD = "admin";
+
+    @Rule
+    public ExternalResource monitorRestEndpoint = new MonitorRestEndpoint();
+
+    @Before
+    public void setup() throws IOException {
+        RestAssured.authentication = basic(USER_NAME, USER_PASSWORD);
+    }
+
+    @Test(timeout = 10 * ONE_MINUTE)
+    public void test() throws Exception {
+        // upload test dictionary
+        final String dictionaryJson = "{ \"name\" :\"DictionaryTest\", \"entries\" : [\"1\", \"5\"],"
+                + " \"caseSensitive\": \"false\"}";
+
+        final String header = given().contentType("application/json").body(dictionaryJson).when().put(REFERENCEDATA_PATH
+                + "DictionaryTest").then().statusCode(HttpStatus.SC_CREATED).extract().header("Location").toString();
+        assertEquals(
+                "http://192.168.99.100:8080/DataCleaner-monitor-ui-5.1.6-SNAPSHOT/repository/demo/referencedata/dictionary/DictionaryTest",
+                header);
+
+        // upload job and set the schedule to be a hot folder()
+        final String jobName = "ReferenceData";
+        final File file = new File("src/test/resources/" + jobName + ".analysis.xml");
+        assertTrue(file.exists());
+        final String command = "docker exec " + HotFolderHelper.getContainerId()
+                + " /bin/sh /tmp/generate-hot-folder-input.sh";
+        HotFolderHelper.getCommandOutput(command);
+
+        final String hotFolder = "/tmp/hot_folder";
+        given().multiPart(file).param("hotfolder", hotFolder).when().post(JOBS_PATH).then().statusCode(
+                HttpStatus.SC_OK);
+
+        // check that the job is scheduled with a hot folder
+        final boolean scheduleCheck = given().contentType("application/json").when().get("/schedules/" + jobName).then()
+                .statusCode(HttpStatus.SC_OK).extract().body().asString().contains(hotFolder);
+
+        assertTrue(scheduleCheck);
+
+        try {
+            // wait for the hot folder trigger and job execution
+            Thread.sleep(20 * 1000);
```

Comment:
I would need the id of the execution, which I don't have since it is automatically triggerred